client/executor/wasmi/src/lib.rs

```diff
@@ -624,30 +624,96 @@ impl StateSnapshot {
 }
 
 /// A runtime along with its initial state snapshot.
-#[derive(Clone)]
 pub struct WasmiRuntime {
+	/// A wasm module.
+	module: Module,
+	/// The host functions registered for this instance.
+	host_functions: Arc<Vec<&'static dyn Function>>,
+	/// Enable stub generation for functions that are not available in `host_functions`.
+	/// These stubs will error when the wasm blob tries to call them.
+	allow_missing_func_imports: bool,
+	/// Numer of heap pages this runtime uses.
+	heap_pages: u64, 
+	/// Data segments created for each new instance.
+	data_segments: Vec<DataSegment>,
```

Comment:
Could this be shared via `Arc` as well? Or do we actually alter it?

Reply:
I'm fairly sure each instance needs to have it's own copy of the data. Right @pepyakin ?

Reply:
I haven't made myself acquainted with the code yet, so here is my general answer:
`data_segments` are extracted from the corresponding `Module` and, like `Module`, doesn't change for a particular version of code.

Reply:
As far as I can see each instantiation of `StateSnapshot` requires an instance of data segments. And each `WasmiInstance` requires a `StateSnapshot`